File: 6_ZigZagConversion/python/ZigZagConversion.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if(numRows<2):
            return s
        sub_str = ["" for i in range(numRows)]
        row_idx = 0
        for i in range(len(s)):
            idx = i%(2*numRows-2)
            if idx<numRows:
                sub_str[idx]+=s[i]
            else:
                idx-=numRows
                sub_str[-idx-2]+=s[i]
        res = ""
        for i in range(len(sub_str)):
            res+=sub_str[i]
        return res

File: 6_ZigZagConversion/c++/ZigZagConversion.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        if(numRows<2){
            return s;
        }
        string res = "";
        vector<string> sub_strs(numRows, "");
        int row_idx;
        for(int i = 0; i<s.size(); i++){
            row_idx = i%(2*numRows-2);
            if(row_idx<numRows){
                sub_strs[row_idx]+=s[i];
            }
            else{
                // row_idx -= numRows;
                row_idx = 2*numRows-row_idx-2;
                sub_strs[row_idx]+=s[i];
            }
        }
        for(int i = 0; i<numRows; i++){
            res+=sub_strs[i];
        }
        return res;
    }
};